### PR TITLE
Added MRW demo, improved usability

### PR DESCRIPTION
**Pull request by MerlinDumeur**
Update so Patrice can get the MRW demo

### CODE REVIEW

File: pymultifracs/mfa.py [hunk 1]
```diff
@@ -63,6 +65,14 @@ def mf_analysis(wt_coefs, wt_leaders, j2_eff, j1, weighted,
         The output of the multifractal analysis
     """
 
+    # In case no value of q is specified, we still include q=2 in order to be
+    # able to estimate H
+    if q is None:
+        q = [2]
+
+    if isinstance(q, list):
```
Comment (philouc) [inline pymultifracs/mfa.py:73]: So do you offer the possibility to enter a list or a np.array for q in the end?

File: pymultifracs/mfspectrum.py [hunk 2]
```diff
@@ -150,7 +150,7 @@ def plot(self, figlabel='Multifractal Spectrum', filename=None):
         plt.plot(self.hq, self.Dq, 'ko-')
         # plt.grid()
         plt.xlabel('h(q)')
-        plt.ylabel('D(q)')
+        plt.ylabel('D(h)')
```
Comment (philouc) [inline pymultifracs/mfspectrum.py:153]: either you put on the x-axis h and on the y-axis D(h) or you put h(q) on the x-label and D(q) on the y-axis, try to be consistent.

File: pymultifracs/utils.py [hunk 3]
```diff
@@ -92,3 +92,33 @@ def linear_regression(x, y, nj, return_variance=False):
         return a, b
     else:
         return a, b, var_a
+
+
+def build_q_log(q_min, q_max, n):
+    """
+    Builds a convenient vector of q values for multifractal analysis
```
Comment (philouc) [inline pymultifracs/utils.py:99]: Build instead of Builds. Very nice code by the way!

File: tests/conftest.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+import pytest
+
+
+@pytest.fixture(scope='session')
+def fbm_generate(tmpdir_factory):
+
+    with open('tests/fbm_config.json', 'rb') as f:
+        config_list = json.load(f)
+
+    for i, config in enumerate(config_list):
```
Comment (philouc) [inline tests/conftest.py:11]: Do you want to print the instance of the simulation like print("Generate fBM process {}".format(i)) before constructing the fBM process

File: tests/simul_test.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+    with pytest.raises(ValueError):
+        fbm(100, H=-1.0)
+    with pytest.raises(ValueError):
+        fbm(100, H=2.0)
+
+
+def test_fbm_shape():
+
+    for shape in [1000, 2345, 4096]:
```
Comment (philouc) [inline tests/simul_test.py:26]: Perhaps try to stick to power-2 lengths

File: tests/simul_test.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    with pytest.raises(ValueError):
+        mrw(100, H=2.0, lam=0.1, L=100)
+
+    with pytest.raises(ValueError):
+        mrw(100, H=0.5, lam=0.1, L=200)
+
+
+def test_mrw_shape():
+
+    for shape in [100, 1000, 4096]:
```
Comment (philouc) [inline tests/simul_test.py:57]: idem: 1024; 2048, 4096

**Review by philouc — CHANGES_REQUESTED** [review]

File: pymultifracs/mfa.py (hunk 1 above)
Comment (MerlinDumeur) [reply pymultifracs/mfa.py:73]: yes, here if q is None, it is set to [2] by default. Then it checks to see if q is a list, and if it is then transforms it into a numpy array. I should probably update the doc to reflect that

Comment (MerlinDumeur) [reply pymultifracs/mfa.py:73]: The reason for [2] by default is so that we can always compute H

File: pymultifracs/mfspectrum.py (hunk 2 above)
Comment (MerlinDumeur) [reply pymultifracs/mfspectrum.py:153]: It was patrice who thought it was better to have D(h) (which is fine). If we then set the x label to h, we lose the information that every point on the spectrum corresponds to a value of q. Maybe we should indicate it somewhere?

File: tests/conftest.py (hunk 4 above)
Comment (MerlinDumeur) [reply tests/conftest.py:11]: This is not yet finished, the idea here is to generate multiple fbms with different parameters which will be used to test all the functions. The idea is to reuse the same fbms for all tests. Insofar as its use will be contained to performing tests, I don't think printing here is necessary, as if all goes well there should be nothing to read, and if things go wrong we will catch them later down the line, in the test functions themselves

File: tests/simul_test.py (hunk 5 above)
Comment (MerlinDumeur) [reply tests/simul_test.py:26]: This test is really fast (basically instant) even without using power-2 lengths, and if not using power-2 lengths causes issues or different behaviour, I'd like to catch it

**Comment by MerlinDumeur** [comment]
This is the last PR I'm going to make in the master branch, the rest will have to be done in different branches as fixes will otherwise get mixed with new features

**Review by philouc — APPROVED** [review]

**Pull request merged**